src/main/java/fi/aalto/cs/apluscourses/model/Course.java

```diff
@@ -57,6 +57,9 @@
   @NotNull
   private final Map<String, URL> resourceUrls;
 
+  @NotNull
+  private final Map<String, String> vmOptions;
```

Comment:
Consider having a separate type for these options instead of just <String, String> pairs.

Reply:
I like the current approach because:
1. it closely resembles the JSON structure (you see Map<> and immediately know how the JSON must look like)
2. it makes it clear that it's a set of key-value pairs. There's nothing more to these options than literally being a set of key-value pairs, so a potential class would just have this one single field and wouldn't actually add anything to it.